src/subapps/projects/components/Activities/ActivityForm.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+import {
+  Form,
+  Input,
+  DatePicker,
+  Radio,
+  Row,
+  Col,
+  Button,
+  Spin,
+  Select,
+} from 'antd';
+import * as moment from 'moment';
+
+import { Status } from '../StatusIcon';
+import { ActivityMetadata } from '../../containers/NewActivityContainer';
+
+import './ActivityForm.less';
+
+const ActivityForm: React.FC<{
+  onClickCancel(): void;
+  onSubmit(data: ActivityMetadata): void;
+  busy: boolean;
+}> = ({ onClickCancel, onSubmit, busy }) => {
+  const [name, setName] = React.useState<string>('');
+  const [nameError, setNameError] = React.useState<boolean>(false);
+  const [description, setDescription] = React.useState<string>('');
+  const [descriptionError, setDescriptionError] = React.useState<boolean>(
+    false
+  );
+  const [summary, setSummary] = React.useState<string>();
+  const [status, setStatus] = React.useState<Status>(Status.toDo);
+  const [dueDate, setDueDate] = React.useState<any>();
+  const [dateError, setDateError] = React.useState<boolean>(false);
+
+  const formItemLayout = {
+    labelCol: { xs: { span: 24 }, sm: { span: 7 } },
+    wrapperCol: { xs: { span: 24 }, sm: { span: 17 } },
+  };
+
+  const isEmptyInput = (value: string) => {
+    return value.split(' ').join('') === '';
+  };
+
+  const isValidInput = () => {
+    let isValid = true;
+
+    if (isEmptyInput(name)) {
+      setNameError(true);
+      isValid = false;
+    } else {
+      setNameError(false);
+    }
+
+    if (isEmptyInput(description)) {
+      setDescriptionError(true);
+      isValid = false;
+    } else {
+      setDescriptionError(false);
+    }
+
+    if (!dueDate) {
+      setDateError(true);
+      isValid = false;
+    } else {
+      setDateError(false);
+    }
+
+    return isValid;
+  };
+
+  const onChangeName = (event: any) => {
+    setName(event.target.value);
+    setNameError(false);
+  };
+
+  const onChangeDescription = (event: any) => {
+    setDescription(event.target.value);
+    setDescriptionError(false);
+  };
+
+  const onChangeDate = (date: any) => {
+    setDueDate(date);
+    setDateError(false);
+  };
+
+  const onClickSubmit = () => {
+    if (isValidInput()) {
+      const data: any = {
+        name,
+        description,
+        summary,
+        dueDate,
+        status,
+      };
+
+      onSubmit(data);
+    }
+  };
+
+  const { Item } = Form;
+
+  return (
+    <Form {...formItemLayout} className="activity-form">
+      <h2 className="activity-form__title">Create New Activity</h2>
+      <Spin spinning={busy} tip="Please wait...">
+        <Row gutter={24}>
+          <Col xs={24} sm={24} md={12}>
```

Comment:
xs={24} sm={24} md={12} can be moved to a variable since we are using the same layout for all columns.